rocAL/rocAL/source/meta_data/bounding_box_graph.cpp

```diff
@@ -162,66 +156,61 @@ void BoundingBoxGraph::update_box_encoder_meta_data(std::vector<float> *anchors,
+                    anchor_xcyxwh.xcycwh.w = ( p_anchor->ltrb.r - p_anchor->ltrb.l ) * scale;      //w
+                    anchor_xcyxwh.xcycwh.h = ( p_anchor->ltrb.b - p_anchor->ltrb.t ) * scale;      //h
                     // Reference for offset calculation between the Ground Truth bounding boxes & anchor boxes in <xc,yc,w,h> format
                     // https://github.com/sgrvinod/a-PyTorch-Tutorial-to-Object-Detection#predictions-vis-%C3%A0-vis-priors
-                    box_bestidx.xc = ((box_bestidx.xc - anchor_xcyxwh.xc) / anchor_xcyxwh.w - means[0]) * inv_stds[0] ;
-                    box_bestidx.yc = ((box_bestidx.yc - anchor_xcyxwh.yc) / anchor_xcyxwh.h - means[1]) * inv_stds[1];
-                    box_bestidx.w = (std::log(box_bestidx.w / anchor_xcyxwh.w) - means[2]) * inv_stds[2];
-                    box_bestidx.h = (std::log(box_bestidx.h / anchor_xcyxwh.h) - means[3]) * inv_stds[3];
+                    box_bestidx.xcycwh.xc = ((box_bestidx.xcycwh.xc - anchor_xcyxwh.xcycwh.xc) / anchor_xcyxwh.xcycwh.w - means[0]) * inv_stds[0] ;
+                    box_bestidx.xcycwh.yc = ((box_bestidx.xcycwh.yc - anchor_xcyxwh.xcycwh.yc) / anchor_xcyxwh.xcycwh.h - means[1]) * inv_stds[1];
+                    box_bestidx.xcycwh.w = (std::log(box_bestidx.xcycwh.w / anchor_xcyxwh.xcycwh.w) - means[2]) * inv_stds[2];
+                    box_bestidx.xcycwh.h = (std::log(box_bestidx.xcycwh.h / anchor_xcyxwh.xcycwh.h) - means[3]) * inv_stds[3];
                     encoded_bb[anchor_idx] = box_bestidx;
-                    encoded_labels[anchor_idx] = bb_labels.at(best_idx);
+                    encoded_labels[anchor_idx] = bb_labels[best_idx];
                 }
                 else
                 {
-                    box_bestidx.xc = 0.5 * (bb_coords[best_idx].l + bb_coords[best_idx].r); //xc
-                    box_bestidx.yc = 0.5 * (bb_coords[best_idx].t + bb_coords[best_idx].b); //yc
-                    box_bestidx.w = bb_coords[best_idx].r - bb_coords[best_idx].l;      //w
-                    box_bestidx.h = bb_coords[best_idx].b - bb_coords[best_idx].t;      //h
+                    box_bestidx.xcycwh.xc = 0.5 * (bb_coords[best_idx].l + bb_coords[best_idx].r); //xc
+                    box_bestidx.xcycwh.yc = 0.5 * (bb_coords[best_idx].t + bb_coords[best_idx].b); //yc
+                    box_bestidx.xcycwh.w = bb_coords[best_idx].r - bb_coords[best_idx].l;      //w
+                    box_bestidx.xcycwh.h = bb_coords[best_idx].b - bb_coords[best_idx].t;      //h
                     encoded_bb[anchor_idx] = box_bestidx;
-                    encoded_labels[anchor_idx] = bb_labels.at(best_idx);
+                    encoded_labels[anchor_idx] = bb_labels[best_idx];
                 }
             }
             else // Not a match
             {
                 if (offset)
                 {
-                    encoded_bb[anchor_idx] = {0, 0, 0, 0};
+                    encoded_bb[anchor_idx].xcycwh = {0, 0, 0, 0};
```

Comment:
Can we just have it as `encoded_bb[anchor_idx]= {0, 0, 0, 0};` Will that work here?